lib/kitchen/driver/rackspace.rb

```diff
@@ -168,19 +184,34 @@ def servicelevel_check(server)
       end
 
       def hostname(server)
-        if config[:servicenet] == false
+        if config[:ssh_network_name]
+          server.addresses[config[:ssh_network_name]].first['addr']
+        elsif config[:connect_public_net] && config[:servicenet] == false
           server.public_ip_address
         else
           server.private_ip_address
         end
       end
 
+      private def public_network
+        # Returns nil if no public ip address should be provisioned
+        '00000000-0000-0000-0000-000000000000' if config[:connect_public_net]
+      end
+
+      private def service_network
```

Comment:
The "Service" net is a Rackspace-specific term.  It means different things in Rackconnect v2 and v3.  Service net and private network are not exact equivalents.  

I am pulling this out of memory, so I could be wrong, but I believe in RCv2, the network on the 10.x.x.x space served by default as a private network among all the instances within the cloud account.  It also served as the "service" net that allowed rackspace automation service and rackconnect services to administer the boxes.  Things like cloud backup and iptables configuration.   In RCv3, the 10-net is no longer a private network among the cloud boxes (at least by default, as far as I know).  It serves solely as a network for Rackspace management services, and if you want a private network you have to add one.